libslide/src/bignum/mul.rs

```diff
@@ -0,0 +1,146 @@
+#![allow(clippy::suspicious_arithmetic_impl)]
+use crate::bignum::utils::convert_poly;
+use crate::bignum::utils::fft;
+use crate::bignum::utils::ifft;
+use crate::bignum::utils::ispowerof2;
+use crate::bignum::utils::recast_user_vec;
+use crate::bignum::utils::round_complex_vec;
+use crate::bignum::Bignum;
+use std::cmp::Ordering;
+use std::ops;
+
+impl ops::Mul for Bignum {
+    type Output = Bignum;
+
+    fn mul(self, rhs: Bignum) -> Bignum {
+        let is_neg: bool = self.is_neg ^ rhs.is_neg;
+        // total length of return vector is p+q+1
+        let total_length = self.dec.len() + self.int.len() + rhs.dec.len() + rhs.int.len();
+        // need self and rhs dec lengths for split later
+        let dec_len = self.dec.len() + rhs.dec.len();
+        // we need to make the vectors the same length at both the decimal and integer level.
+        let (mut lhs_dec, mut rhs_dec) = match self.dec.len().cmp(&rhs.dec.len()) {
+            Ordering::Greater => {
+                let mut rhs_dec = vec![0; self.dec.len()];
+                for (i, e) in rhs.dec.iter().enumerate() {
+                    rhs_dec[i] = *e
+                }
```

Comment:
I think it would be better to reserve `self.dec.len() - rhs.dec.len()` more memory in `rhs.dec` and fill the new memory with 0s.